### DISCUSSION TITLE
How Restore config will work.

### DISCUSSION THREAD

**Discussion by kRHYME7** [kRHYME7/Hyde-cli#12] [Show and tell]
Okay, I decided something. This restore file will be inside ``` ~/.config/hyprdots-ctl/ ```  (I will move some config files to this directory too so that it will be Isolated from the ~/.config/hypr directory )
First, we need to realize these 3 directories 

+ Clone directory is the directory that has the Configs (forked, upstream, or a backup)
+ Target directory is your current configuration 
+ Backup Directory is the destination of backups (Think of a compilation of files that will be overwritten)

Now flags are used to determine what you do to the files:

**O** for overwrite so this one is it moves the target files/directory to the backup directory and then copies the Clone Directory files to the target directory. This will effectively overwrite the files and now your current files will be in the backup

**P or any** flag for populate,  so this one if not declared will do something like copy Clone configs to the Target if only the files on the Target files do not exist. This is helpful for Fresh installations.  

**B** is for Backup so this will just recursively backup your configs. So target files will be  copied to the backup directory nothing else

**S** stands for Sync so this flag copies your target directory to backup, then ``` cp -rf ```/copies the Clone directory to the target directory. Now S flags are still guaranteed to overwrite some of the configs but will not overwrite the entire directory only the ones that have the same filename. Effectively leaving your changes intact(for example You have modules on your waybar or user wallpapers in the swww folder ) 

** I ** flags will only be ignored and skipped so that we can have a more declarative approach Having this will break some functionality. But maybe useful. Same as just deleting the whole line.

Of course, this will be integrated with these commands 
```
❯ Hyprdots restore
                          [defaults]       Restore to Default Dots of Hyprdots
                          [Clone]          Restores Dots from a Clone Repository
                          [BackUp]         Restores Dots to a previous state.
❯ Hyprdots backup
                          [all]            Backs up the current configuration.
                          [revert]         Revert to a Previous Backup run by 'Hyprdots backup all'.
                          [list]           List the backup
                          [clean]          Clear some Storage by Deleting old backups. 
```

**Comment by Enigma1309** [comment]
> This restore file will be inside ~/.config/hyprdots-ctl/

hey i think this needs to be updated...
since hyde no longer reads `~/.config/hyprdots-ctl/manage_cfg.lst` 
it only seems to read file present inside `.cache/hyde/manage_cfg.lst` is it a bug? or there's any other place i have to edit this file?

**Comment by kRHYME7** [comment]
ops this is outdated.

`Hyde restore Config ` will just save it for you.

**Reply by Enigma1309** [reply]
> is there any other place i have to edit this file?

apart from `~/.cache/hyde/manage_cfg.lst`
could you add the `~/.config/hyde/manage_cfg.lst` and then symlink this file to the one present in `~/.cache/hyde/manage_cfg.lst`?

**Reply by kRHYME7** [reply]
You want to add a symlink of the file to ~/.config/hyde/ ?

**Reply by Enigma1309** [reply]
i want hyde-cli to store it's manage_cfg.lst somewhere other than in .cache dir,
maybe in `~/.config/hyde/manage_cfg.lst` and this should be crated by default (you may symlink this file in cache dir, so that you don't have to change a bunch of code)

**Reply by kRHYME7** [reply]
uhmm This should be in cache dir as the CLI reads the changes of restore_cfg.lst, when there's a modified restore_cfg.lst the manage_cfg.lst will reset. 
This is not intended to be edited manually. Tittu and I decided to move it to cache to have a cleaner ~/.config/hyde/ directory.


Do you want  to add your custom  lines for manage_cfg.lst? eg. adding new files/dirs?

What are your use cases for this file? 

Note:  a copy of manage_cfg.lst will always be stored on the backup Directory so you can reuse it.
This way the CLI doesn't need to  re interpret the contents of the backup Directory as the manage_cfg.lst is already there.

**Comment by Enigma1309** [comment]
> Do you want to add your custom lines for manage_cfg.lst? eg. adding new files/dirs?

yeah i do

> This should be in cache dir as this reads the changes of restore_cfg.lst, when there's a modified restore_cfg.lst the manage_cfg.lst will reset.

i can technically edit `~/Hyprdots/Scripts/restore_cfg.lst` but that causes the issue in git pull and i've to resolve it....
rather i think it would be a good idea if we have different config managing list which is outside of Hyprdots directory
say `x.lst` and when Hyde will read this it will intersect the the restore_cfg.lst and `x.lst` and this way we can manage our own changes ....
> What are your use cases for this file?

i basically want to keep my configs consistent with rolling updates

**Reply by kRHYME7** [reply]
It's possible to declare it in hyde.conf.  Are you using a fork? the list will be skipped if the files do not exist inside ./Configs

**Reply by Enigma1309** [reply]
wait how?? also nah i'm not on fork

**Reply by kRHYME7** [reply]
I meant we can add that option later. But what else are you planning to add?

**Reply by Enigma1309** [reply]
i wanna add these and a few more
![image](https://github.com/kRHYME7/Hyde-cli/assets/94917588/160145bd-1e47-42f6-b610-f34f736d99fe)

**Reply by kRHYME7** [reply]
Ops that won't work because the `... waybar|modules|waybar  ... ` is treted as a directory. 

BTW, I think hyde.conf is not included in the restore list?

**Reply by Enigma1309** [reply]
so what am i doing wrong?

**Reply by kRHYME7** [reply]
You're declaring the files inside modules. But modules is already processed above it. 

See this PR https://github.com/prasanthrangan/hyprdots/pull/1418

this will grant as a capability to add a #custom module

 We will use clock module for example:

so we make a separate file ` ./clock#myown.jsonc `  then declare it inside `  ./.config.ctl ` like ` ... clock#myown `

**Reply by Enigma1309** [reply]
okay
i'll wait for it

**Reply by kRHYME7** [reply]
So that we can just preserve the  ` config.ctl ` file then everything will be up to date + all your modules can be preserved. 


 Actually, the only change you need is this lol   ![image](https://github.com/kRHYME7/Hyde-cli/assets/53417443/c1a3eefd-8925-4e3f-ade0-018652fe485b)

**Reply by Enigma1309** [reply]
okay done

**Reply by kRHYME7** [reply]
Now you can just add config.ctl to preserve, then module directory to sync. 

![image](https://github.com/kRHYME7/Hyde-cli/assets/53417443/ccb56be9-df0c-4c09-9569-1d607badcf27)

**Comment by Enigma1309** [comment]
Hey thanks for all of that but should i be editing `~/.cache/hyde/manage_cfg.lst `
or other file which will serve the same purpose?

**Reply by kRHYME7** [reply]
Just don't touch those files. ` Hyde restore Config ` lets you edit and save it. Would it be fine if we add a command to modify the control file?

You can just change it later if you want to update

**Reply by Enigma1309** [reply]
> Would it be fine if we add a command to modify the control file?

yeah that's what i want :)

**Reply by kRHYME7** [reply]
Post an issue please, I might forget.

**Reply by Enigma1309** [reply]
okay
